### PR TITLE
OperatingSystem.IsAndroid()

### PR DESCRIPTION
**Pull request by SupinePandora43**
Adds NET 5.0 target framework

uses `OperatingSystem.IsAndroid` for android checks

### CODE REVIEW

**Comment by zaafar** [comment]
what's up with so many white space changes. :D

**Comment by SupinePandora43** [comment]
> what's up with so many white space changes. :D

Vscode automatically normalized line endings using `.editorconfig`

**Comment by SupinePandora43** [comment]
this is not needed if https://github.com/dotnet/runtime/issues/51052 will be fixed (when `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)` will return `true`)

**Review by TechnologicalPizza — COMMENTED** [review]
Seems like we may want to deprecate the old NativeLibraryLoader as it breaks horribly for publishes with embedded native dlls.

File: src/Veldrid/Vk/VkGraphicsDevice.cs [hunk 1]
```diff
@@ -1,1519 +1,1523 @@
...
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (availableInstanceExtensions.Contains(CommonStrings.VK_KHR_WIN32_SURFACE_EXTENSION_NAME))
+                {
+                    _surfaceExtensions.Add(CommonStrings.VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
```
Comment (TechnologicalPizza) [inline src/Veldrid/Vk/VkGraphicsDevice.cs:471]: We may want to change the use of `RuntimeInformation` throughout the whole library at once. The solution here is not future-proof (checking for NET5_0 will not work for NET6).
We may just want to create a NET6 branch of Veldrid 🤔

Comment (SupinePandora43) [reply src/Veldrid/Vk/VkGraphicsDevice.cs:471]: `OperatingSystem.IsAndroid()` was introduced in net5, and works as expected in net6.
In other words: net5 added, net6 implemented.

Comment (TechnologicalPizza) [reply src/Veldrid/Vk/VkGraphicsDevice.cs:471]: Well I meant the #if directive in particular. I'm not sure the `NET5_0` symbol will be available in NET6 (the `NET` symbol should exist though)

Comment (SupinePandora43) [reply src/Veldrid/Vk/VkGraphicsDevice.cs:471]: `NET5_0_OR_GREATER`

File: src/Veldrid/MTL/MTLGraphicsDevice.cs [hunk 2]
```diff
@@ -8,6 +8,10 @@
 using NativeLibraryLoader;
 using Veldrid.MetalBindings;
 
+#if NET5_0_OR_GREATER
+using NativeLibrary = NativeLibraryLoader.NativeLibrary;
```
Comment (mellinoe) [inline src/Veldrid/MTL/MTLGraphicsDevice.cs:12]: I don't think this needs to be behind an `#if`. It should be harmless to include everywhere.

File: src/Veldrid/MTL/MTLGraphicsDevice.cs [hunk 3]
```diff
@@ -16,7 +20,7 @@ internal unsafe class MTLGraphicsDevice : GraphicsDevice
         private static readonly Dictionary<IntPtr, MTLGraphicsDevice> s_aotRegisteredBlocks
             = new Dictionary<IntPtr, MTLGraphicsDevice>();
 
-        private readonly MTLDevice _device;
+        private readonly MTLDevice _device;
```
Comment (mellinoe) [inline src/Veldrid/MTL/MTLGraphicsDevice.cs:23]: There's a lot of unnecessary whitespace changes included in this diff. I can't tell what was actually changed. Can you remove these unexpected diffs from the PR?

File: src/Veldrid/MTL/MTLGraphicsDevice.cs [hunk 4]
```diff
@@ -8,6 +8,10 @@
 using NativeLibraryLoader;
```
Comment (mellinoe) [inline src/Veldrid/MTL/MTLGraphicsDevice.cs:8]: I think this may not be needed if the addition below is made everywhere (instead of behind the #if).

**Review by mellinoe — CHANGES_REQUESTED** [review]
The functional changes look fine, please clean up some of the misc. things I commented on and we can merge.

File: src/Veldrid/MTL/MTLGraphicsDevice.cs (hunk 3 above)
Comment (SupinePandora43) [reply src/Veldrid/MTL/MTLGraphicsDevice.cs:23]: The previous attempt had much more space changes because editor used `.editorconfig`

Comment (SupinePandora43) [reply src/Veldrid/MTL/MTLGraphicsDevice.cs:19]: @mellinoe i can't
image
^ this is what i get when trying to edit files on `master` branch.
those space changes are auto fixes by editor

**Comment by mellinoe** [comment]
I merged this after rebasing it on master. Thanks for the contribution!

**Pull request closed without merging**